### PR TITLE
Interaction - Add crew status check for some vehicle interactions

### PR DESCRIPTION
**Pull request by johnb432**
**When merged this pull request will:**
- If one or more passengers in a vehicle are conscious, not captive and hostile to the interacting unit, interactions such as ACE passenger and cargo actions will not appear.
- Updated `FUNC(addPassengerActions)`. Using `objectParent` instead of `vehicle`.
- Addresses https://github.com/acemod/ACE3/pull/9618#issuecomment-1796694348.

I opted not to go with adding a condition for interacting via `EFUNC(common,addCanInteractWithCondition)`, as I don't want to impose this for all vehicle interactions.

### IMPORTANT

- If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

**Comment by LinkIsGrim** [comment]
Haven't tested yet. How does this work with empty vehicles?

File: addons/interaction/functions/fnc_canInteractWithVehicleCrew.sqf [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * Example:
+ * [cursorObject, player] call ace_interaction_fnc_canInteractWithVehicleCrew
+ *
+ * Public: No
+ */
+
+params ["_player", "_vehicle"];
+
+private _sidePlayer = side _player;
```
Comment (LinkIsGrim) [inline addons/interaction/functions/fnc_canInteractWithVehicleCrew.sqf:21]: `side group _player`?

Comment (johnb432) [reply addons/interaction/functions/fnc_canInteractWithVehicleCrew.sqf:21]: I figured we would never need to know the real side of the player if you are using this function. However I don't think it hurts and it's safer.

**Comment by johnb432** [comment]
> Haven't tested yet. How does this work with empty vehicles?

It works as expected.

File: addons/interaction/functions/fnc_canInteractWithVehicleCrew.sqf [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * Public: No
+ */
+
+params ["_player", "_vehicle"];
+
+private _sidePlayer = side group _player;
+
+((crew _vehicle) select {_x != _player && {!unitIsUAV _x}}) findIf { // ignore player and UAV units
+    // Units must all be unconscious, captive or friendly (side group is used in case unit is captive/unconscious) for actions to show up
+    !captive _x && {lifeState _x in ["HEALTHY", "INJURED"]} && {[_sidePlayer, side group _x] call BIS_fnc_sideIsEnemy}
```
Comment (LinkIsGrim) [inline addons/interaction/functions/fnc_canInteractWithVehicleCrew.sqf:25]: Not exactly related but we should update common `FUNC(isAwake)` to do this lifeState check instead, methinks.

File: addons/interaction/functions/fnc_canInteractWithVehicleCrew.sqf [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * Unit can interact with vehicle crew <BOOL>
+ *
+ * Example:
+ * [cursorObject, player] call ace_interaction_fnc_canInteractWithVehicleCrew
+ *
+ * Public: No
+ */
+
+params ["_player", "_vehicle"];
+
```
Comment (LinkIsGrim) [inline addons/interaction/functions/fnc_canInteractWithVehicleCrew.sqf:20]: Worth adding early exit for empty vehicle maybe?

**Review by LinkIsGrim — APPROVED** [review]
See early exit comment, but it's good as is.

File: addons/interaction/functions/fnc_canInteractWithVehicleCrew.sqf (hunk 2 above)
Comment (johnb432) [reply addons/interaction/functions/fnc_canInteractWithVehicleCrew.sqf:25]: Generally speaking, I think we should change `_unit getVariable ["ACE_isUnconscious", false]` to `lifeState` (or `EFUNC(common,isAwake)`, if we update it as per your suggestion) wherever possible.

Comment (LinkIsGrim) [reply addons/interaction/functions/fnc_canInteractWithVehicleCrew.sqf:25]: Agreed.

**Review by LinkIsGrim — APPROVED** [review]

**Pull request merged**